lib/Attean/Endpoint.pm

```diff
@@ -136,15 +137,19 @@ package Attean::Endpoint 0.010 {
 	
 	sub BUILDARGS {
 		my $class		= shift;
+		my @params = @_;
 		my %args;
-		if (blessed($_[0]) and $_[0]->does('Attean::API::Model')) {
+		if (blessed($params[0]) and $params[0]->does('Attean::API::Model')) {
 			# ->new( $model, \%conf )
-			$args{ model }	= shift;
-			$args{ conf }	= shift;
+			$args{ model }	= shift @params;
+			$args{ conf }	= shift @params;
 			$args{ graph }	= Attean::IRI->new('http://example.org/graph');
+		} elsif (any { blessed($_) && $_->does('Attean::API::Model') } @params) {
+			# Assume the buildargs can be taken directly
```

Comment:
What's the expected API to call this? Something like `->new( model => $model )`?